File: sys/src/nonlinear/levenberg_marquardt_params.cc
#include "levenberg_marquardt_params.h"

namespace gtsam {

std::unique_ptr<LevenbergMarquardtParams> default_levenberg_marquardt_params() {
  return std::make_unique<LevenbergMarquardtParams>();
}

void levenberg_marquardt_params_set_max_iterations(
    LevenbergMarquardtParams &params, uint32_t value) {
  return params.setMaxIterations(value);
}

} // namespace gtsam

File: src/nonlinear/levenberg_marquardt_params.rs
use cxx::UniquePtr;

pub struct LevenbergMarquardtParams {
    pub(super) inner: UniquePtr<::sys::LevenbergMarquardtParams>,
}

impl Default for LevenbergMarquardtParams {
    fn default() -> Self {
        Self {
            inner: ::sys::default_levenberg_marquardt_params(),
        }
    }
}

impl LevenbergMarquardtParams {
    pub fn set_max_iterations(&mut self, value: u32) {
        ::sys::levenberg_marquardt_params_set_max_iterations(self.inner.pin_mut(), value)
    }
}
